modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -244,17 +250,18 @@ def _simulate_and_detect_failure(self, servers, failed_nodes, timeout):
 
     def _check_failed_number(self, failed_nodes, survived_node):
         """Ensures number of failed nodes is correct."""
-        cmd = "grep '%s' %s | wc -l" % (node_failed_event_pattern(), IgniteAwareService.STDOUT_STDERR_CAPTURE)
+        failed_cnt = int(exec_command(survived_node, "grep '%s' %s | wc -l" %
+                                      (node_failed_event_pattern(), IgniteAwareService.STDOUT_STDERR_CAPTURE)))
 
-        failed_cnt = int(str(survived_node.account.ssh_client.exec_command(cmd)[1].read(), sys.getdefaultencoding()))
+        # Cache survivor id, do not read each time.
+        surv_id = read_node_id(survived_node)
```

Comment:
could this be IgniteAppService method or property?

Reply:
Moved to the ignite utils